app/views/consignment_references.scala.html

```diff
@@ -35,25 +35,37 @@
         @components.error_summary(form.errors)
 
         @components.heading(
-            title = messages("consignmentReferences.question"),
+            title = messages(s"consignmentReferences.${request.choice.value}.question"),
             sectionHeader = messages(s"consignmentReferences.${request.choice.value}.heading")
         )
 
         @components.input_radio(
           field = form("reference"),
             legend = "",
             inputs = Seq(
-              RadioOption("Ducr", Ducr, "consignmentReferences.reference.ducr"),
-              RadioOption("Mucr", Mucr, "consignmentReferences.reference.mucr")
+              RadioOption(
+                  id = "Ducr",
+                  value = Ducr,
+                  messageKey = "consignmentReferences.reference.ducr",
+                  revealComponent = Some(components.input_text(
+                      field = form("ducrValue"),
+                      label = messages("consignmentReferences.reference.ducrValue"),
+                      labelClass = Some("bold-small")
+                  ))
+              ),
+              RadioOption(
+                  id = "Mucr",
+                  value = Mucr,
+                  messageKey = "consignmentReferences.reference.mucr",
+                  revealComponent = Some(components.input_text(
+                      field = form("mucrValue"),
```

Comment:
Hmm, I think that instead of creating models with two values etc, we can in both `revealComponent` put input text that will point to the same form field.

So if we will stay with a model and everything here we can add `form("referenceValue")` and in line 53 the same. Both inputs will be pointing to the same field.

Reply:
Is it possible and will it be working correctly? Or you tried it before and it's not working?